### PR TITLE
Fixed a memory leak and bad free.

### PR DESCRIPTION
**Pull request by akallabeth**
Fixed issues introduced with #3864

### CODE REVIEW

**Comment by akallabeth** [comment]
@hardening Looks like we missed a leak :/ This one should fix it (clang scanbuild confirms). Just be careful if you rerun, due to the ```INVALID_HANDLE_VALUE``` check it now reports a false positive (or better a rare case where there could actually be a memory leak if the address is ```-1```)

**Comment by akallabeth** [comment]
@mfleisz @bmiklautz @nfedera Anyone time to have a look?

File: winpr/libwinpr/file/generic.c [hunk 1]
```diff
@@ -475,7 +475,7 @@ BOOL WINAPI GetFileAttributesExA(LPCSTR lpFileName, GET_FILEEX_INFO_LEVELS fInfo
 	WIN32_FIND_DATAA findFileData;
 	HANDLE hFind;
 
-	if ((hFind = FindFirstFileA(lpFileName, &findFileData) != INVALID_HANDLE_VALUE))
+	if ((hFind = FindFirstFileA(lpFileName, &findFileData)) != INVALID_HANDLE_VALUE)
```
Comment (nfedera) [inline winpr/libwinpr/file/generic.c]: nice finding! however, return false on error is required too

Comment (akallabeth) [inline winpr/libwinpr/file/generic.c]: @nfedera done.

**Comment by akallabeth** [comment]
@freerdp-bot test please

**Pull request merged**